contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -2130,6 +2130,63 @@ class tsqlBuilder : public tsqlCommonMutator
 		handleFloatWithoutExponent(ctx);
 	}
 
+	void exitPredicate(TSqlParser::PredicateContext *ctx) override
+	{
+		// For comparison operators directly followed by an '@@' variable, insert a space 
+		// to avoid these being parsed incorrectly by PG; this applies to the following 
+		// character sequences: =@@, >@@, <@@ as well as !=@ (i.e. a single-@ variable)
+		// Note: this issue does not occur for assignments like 'SET @v=@@spid' or column aliases like 'SELECT a=@@spid'
+
+		if ((ctx->comparison_operator()) && (ctx->expression().size() > 1))
+		{
+			std::string op = getFullText(ctx->comparison_operator());		
+			if ((op.back() == '=') || 
+			    (op.back() == '>') || 
+			    (op.back() == '<'))
+			{
+				// The operator must be followed immediately by the variable without any character in between
+				Assert(ctx->expression(1));
```

Comment:
Would you mind assert for ctx->expression(1)->start too ?

Reply:
start and stop always have a value for a node/token. if we cannot rely on that, we cannot rely on ANTLR

Reply:
To illustrate, start and stop are not explicitly checked for  a non-null value anywhere in the code.